tests/models/test_models_unet_3d_condition.py

```diff
@@ -190,23 +198,240 @@ def test_model_attention_slicing(self):
             output = model(**inputs_dict)
         assert output is not None
 
-    # (`attn_processors`) needs to be implemented in this model for this test.
-    # def test_lora_processors(self):
+    def test_lora_processors(self):
+        init_dict, inputs_dict = self.prepare_init_args_and_inputs_for_common()
+
+        init_dict["attention_head_dim"] = 8
+
+        model = self.model_class(**init_dict)
+        model.to(torch_device)
+
+        with torch.no_grad():
+            sample1 = model(**inputs_dict).sample
+
+        lora_attn_procs = {}
```

Comment:
We can leverage the `create_lora_layers()` here, no?

Reply:
Agree - @Pie31415 could we maybe as a final todo factor out this code: 

```py
        for name in model.attn_processors.keys():
            has_cross_attention = name.endswith("attn2.processor") and not (
                name.startswith("transformer_in") or "temp_attentions" in name.split(".")
            )
            cross_attention_dim = model.config.cross_attention_dim if has_cross_attention else None

            if name.startswith("mid_block"):
                hidden_size = model.config.block_out_channels[-1]
            elif name.startswith("up_block"):
                block_id = int(name[len("up_blocks.")])
                hidden_size = list(reversed(model.config.block_out_channels))[block_id]
            elif name.startswith("down_blocks"):
                block_id = int(name[len("down_blocks.")])
                hidden_size = model.config.block_out_channels[block_id]
            elif name.startswith("transformer_in"):
                # Note that the `8 * ...` comes from: https://github.com/huggingface/diffusers/blob/7139f0e874f10b2463caa8cbd585762a309d12d6/src/diffusers/models/unet_3d_condition.py#L148
                hidden_size = 8 * model.config.attention_head_dim

            lora_attn_procs[name] = LoRAAttnProcessor(hidden_size=hidden_size, cross_attention_dim=cross_attention_dim)

            with torch.no_grad():
                lora_attn_procs[name].to_q_lora.up.weight += 1
                lora_attn_procs[name].to_k_lora.up.weight += 1
                lora_attn_procs[name].to_v_lora.up.weight += 1
                lora_attn_procs[name].to_out_lora.up.weight += 1
```

into a `create_lora_layers()` as it's used three times further below? :-) 

Think after this we're good for merge :heart: